pkgdb2/api/admin.py

```diff
@@ -0,0 +1,355 @@
+    '''
+Get a specific Admin Action
+---------------------------
+    Return the desired Admin Action using its identifier.
+
+    ::
+
+        /admin/action/<actionid>
+        /admin/action/?actionid=<actionid>
+
+    Accept GET queries only.
+
+    :arg actionid: An integer representing the identifier of the admin
+        action in the database. The identifier is returned in the
+        API, see ``List admin actions``.
+
+    Sample response:
+
+    ::
+
+        {
+          "action": "request.branch",
+          "collection": {
+            "branchname": "epel7",
+            "dist_tag": ".el7",
+            "koji_name": "epel7",
+            "name": "Fedora EPEL",
+            "status": "Active",
+            "version": "7"
+          },
+          "date_created": 1410161489.0,
+          "date_updated": 1410168952.0,
+          "from_collection": {
+            "branchname": "master",
+            "dist_tag": ".fc21",
+            "koji_name": "rawhide",
+            "name": "Fedora",
+            "status": "Under Development",
+            "version": "devel"
+          },
+          "id": 1,
+          "info": {},
+          "package": {
+            "acls": [],
+            "creation_date": 1397204290.0,
+            "description": null,
+            "name": "R-BiocGenerics",
+            "review_url": null,
+            "status": "Approved",
+            "summary": "Generic functions for Bioconductor",
+            "upstream_url": null
+          },
+          "status": "Approved",
+          "user": "pingou"
+        }
+
+    '''
+    httpcode = 200
+    output = {}
+
+    actionid = flask.request.args.get('actionid', actionid)
+
+    admin_action = pkgdblib.get_admin_action(SESSION, actionid)
+    if not admin_action:
+        output['output'] = 'notok'
+        output['error'] = 'No Admin action with this identifier found'
+        httpcode = 500
+    else:
+        output = admin_action.to_json()
+        output['output'] = 'ok'
+
+    jsonout = flask.jsonify(output)
+    jsonout.status_code = httpcode
+    return jsonout
+
+
+@API.route('/admin/action/status', methods=['POST'])
+@is_admin
+def api_admin_action_edit_status():
+    '''
+Edit Admin Action status update
+-------------------------------
+    Edit the status of an Admin Action.
+
+    ::
+
+        /admin/action/status
+
+    Accept POST queries only.
+
+    :arg id: An integer representing the identifier of the admin
+        action to update in the database. The identifier is returned in the
+        API, see ``List admin actions``.
+    :arg status: The status to which the action should be updated.
+        Can be any of: ``Approved``, ``Awaiting Review``, ``Blocked``,
+        ``Denied``, ``Pending`` and ``Obsolete``.
+
+    ..note:: The ``Obsolete`` status can only be set by the person who made
+             the request.
+
+    Sample response:
+
+    ::
+
+        {
+          "output": "ok",
+          "messages": ["Admin action status updated to: Approved"]
```

Comment:
>    Again with the full sentences in response codes?
>    I would prefer machine-interpretable responses so you could for example
>    translate the API client without having to do manual string fiddling.

We have had this discussion before and while I agree with you, for the moment I
keep the behavior consistent with the rest of the API